engine/src/main/java/org/terasology/engine/subsystem/headless/renderer/HeadlessWorldRenderer.java

```diff
@@ -300,4 +295,15 @@ public int compare(RenderableChunk o1, RenderableChunk o2) {
         }
     }
 
+	@Override
+	public RenderingConfig getRenderingConfig() {
```

Comment:
Ok, meanwhile I found a bit more about injection.

Let's assume you want a PerspectiveCamera instance to have a worldProvider injected in it.

Within PerspectiveCamera you just write:

```java
@In 
private WorldProvider worldProvider.
```

Within the class that instantiates the camera (WorldRendererImpl in our case) you have to write something like:

```java
Camera myCamera = new PerspectiveCamera();
InjectionHelper.inject(myCamera, context);
```

The second line will fill the worldProvider field with the instance it finds in the context - the context is basically a <Class, Instance> map. Notice that in this scenario the @In annotated fields will be null on construction. So, if something needs to be done before the camera can be used, it will need a method called after injection to do it, i.e. an initialise() method. 

So, as far as I can see, you'd save a few lines of code if you just gave the context directly to the constructor and initialized the fields within it instead of using the @In annotations - I'm asking confirmation of this.